database.c

```diff
@@ -742,6 +751,10 @@ database_init(void)
   custom_work  = xstrfmt("%s.tmp", custom_path);
   custom_back  = xstrfmt("%s.bak", custom_path);
 
+  if (access(current_path, F_OK) != 0)
+  {
```

Comment:
It may make sense to use glibs file_test here instead.

Reply:
Possibly, to test for dir not just existence, but the rest of the code uses access() for this stuff

Reply:
Since glib also use access this is more a style thing in this case. I prefer which works better.

Reply:
No opinion on access vs. glib. Just thought I should point out that it's not testing for the directory, but actually the main config file ( $HOME/.config/profiled/current ). I just thought it was easier, since that value was already sitting in a variable for me, and functionally it seemed more or less equivalent...

Reply:
IMO access is just fine for the purpose. There is code else where to make sure it is a directory.

But what is missing dealing with (future) clean devices where the directory structure gets created later on when saving data i.e. migration is needed only when new directory does not exist and old directory does exist -> both need to be checked to avoid unnecessary migration attempt that would probably make some noise.